src/uinput.rs

```diff
@@ -188,3 +232,78 @@ impl VirtualDevice {
         self.write_raw(&[syn])
     }
 }
+
+/// This struct is returned from the [VirtualDevice::enumerate_dev_nodes] function and will yield
+/// the syspaths corresponding to the virtual device. These are of the form `/dev/input123`.
+#[cfg(not(feature = "tokio"))]
```

Comment:
Bit of a further nit, but can we either ungate this or gate this behind a `blocking` feature? In a single binary if one crate user were to define `tokio` and another does not it'll really confuse cargo. I'd like to make sure that if someone really wants to they could build and use both simultaneously.

Reply:
Oh right, it makes sense to not even have the feature gate there if we are differentiating them anyway.

Reply:
You will need to somehow disambiguate the two `DevNodes` structs as well.

Reply:
That's right. I had a much better look now and fixed a few other things too:
 - Added `fs` feature flag to the tokio dependency, so `cargo b --features tokio` actually works.
 - Because of the `.await?`, the async version of the iterator is dealing with just `Option<T>` which simplifies the code.
 - Renamed the `DevNodes` variant for the sync version to `DevNodesBlocking` so they can co-exist.
 
 With the tokio feature flag, one should have access to both the sync and async versions.